File: app/src/main/java/net/ktnx/mobileledger/db/AccountValue.java
/*
 * Copyright © 2021 Damyan Ivanov.
 * This file is part of MoLe.
 * MoLe is free software: you can distribute it and/or modify it
 * under the term of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your opinion), any later version.
 *
 * MoLe is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License terms for details.
 *
 * You should have received a copy of the GNU General Public License
 * along with MoLe. If not, see <https://www.gnu.org/licenses/>.
 */

package net.ktnx.mobileledger.db;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;
import androidx.room.PrimaryKey;


@Entity(tableName = "account_values", indices = {
        @Index(name = "un_account_values", unique = true, value = {"account_id", "currency"}),
        @Index(name = "fk_account_value_acc", value = "account_id")
}, foreignKeys = {
        @ForeignKey(entity = Account.class, parentColumns = "id", childColumns = "account_id",
                    onDelete = ForeignKey.CASCADE, onUpdate = ForeignKey.RESTRICT)
})
public class AccountValue {
    @ColumnInfo
    @PrimaryKey(autoGenerate = true)
    long id;
    @ColumnInfo(name = "account_id")
    private long accountId;
    @NonNull
    @ColumnInfo(defaultValue = "")
    private String currency = "";
    @ColumnInfo
    private float value;
    @ColumnInfo(defaultValue = "0")
    private long generation = 0;
    @Nullable
    @ColumnInfo(name = "amount_style")
    private String amountStyle;
    public long getId() {
        return id;
    }
    public void setId(long id) {
        this.id = id;
    }
    public long getAccountId() {
        return accountId;
    }
    public void setAccountId(long accountId) {
        this.accountId = accountId;
    }
    @NonNull
    public String getCurrency() {
        return currency;
    }
    public void setCurrency(@NonNull String currency) {
        this.currency = currency;
    }
    public float getValue() {
        return value;
    }
    public void setValue(float value) {
        this.value = value;
    }
    public long getGeneration() {
        return generation;
    }
    public void setGeneration(long generation) {
        this.generation = generation;
    }
    @Nullable
    public String getAmountStyle() {
        return amountStyle;
    }
    public void setAmountStyle(@Nullable String amountStyle) {
        this.amountStyle = amountStyle;
    }
}

File: app/src/main/kotlin/net/ktnx/mobileledger/db/AccountValue.kt
/*
 * Copyright © 2021 Damyan Ivanov.
 * This file is part of MoLe.
 * MoLe is free software: you can distribute it and/or modify it
 * under the term of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your opinion), any later version.
 *
 * MoLe is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License terms for details.
 *
 * You should have received a copy of the GNU General Public License
 * along with MoLe. If not, see <https://www.gnu.org/licenses/>.
 */

package net.ktnx.mobileledger.db

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "account_values",
    indices = [
        Index(name = "un_account_values", unique = true, value = ["account_id", "currency"]),
        Index(name = "fk_account_value_acc", value = ["account_id"])
    ],
    foreignKeys = [
        ForeignKey(
            entity = Account::class,
            parentColumns = ["id"],
            childColumns = ["account_id"],
            onDelete = ForeignKey.CASCADE,
            onUpdate = ForeignKey.RESTRICT
        )
    ]
)
class AccountValue {
    @ColumnInfo
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0

    @ColumnInfo(name = "account_id")
    var accountId: Long = 0

    @ColumnInfo(defaultValue = "")
    var currency: String = ""

    @ColumnInfo
    var value: Float = 0f

    @ColumnInfo(defaultValue = "0")
    var generation: Long = 0

    @ColumnInfo(name = "amount_style")
    var amountStyle: String? = null
}
